crates/miden-agglayer/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec;
+use alloc::vec::Vec;
+
+use miden_assembly::Library;
+use miden_assembly::utils::Deserializable;
+use miden_core::Program;
+use miden_protocol::account::{AccountComponent, StorageSlot};
+use miden_utils_sync::LazyLock;
+
+pub mod errors;
+pub mod utils;
+
+// AGGLAYER NOTE SCRIPTS
+// ================================================================================================
+
+// Initialize the B2AGG note script only once
+static B2AGG_SCRIPT: LazyLock<Program> = LazyLock::new(|| {
+    let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/B2AGG.masb"));
+    Program::read_from_bytes(bytes).expect("Shipped B2AGG script is well-formed")
+});
+
+/// Returns the B2AGG (Bridge to AggLayer) note script.
+pub fn b2agg_script() -> Program {
+    B2AGG_SCRIPT.clone()
+}
+
+// AGGLAYER ACCOUNT COMPONENTS
+// ================================================================================================
+
+// Initialize the unified AggLayer library only once
+static AGGLAYER_LIBRARY: LazyLock<Library> = LazyLock::new(|| {
+    let bytes =
+        include_bytes!(concat!(env!("OUT_DIR"), "/assets/account_components/agglayer.masl"));
+    Library::read_from_bytes(bytes).expect("Shipped AggLayer library is well-formed")
+});
```

Comment:
Not for this PR, but it feels a bit weird that we are building a library out of components. In other crates, we do it the other way around. We could probably use the following structure here as well:

```
asm
├── bridge
│   ├── bridge_in.masm
│   ├── bridge_out.masm
│   ├── local_exit_tree.masm
│   ├── asset_conversion.masm
│   └── faucet_registry.masm
├── faucet
├── components
└── note_scripts
```

And we don't have to have all of these be components. For example, asset conversions are just utility procedures that can be used by different components.